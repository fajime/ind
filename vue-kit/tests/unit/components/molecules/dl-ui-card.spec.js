import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';

import DlUicard from '../../../../src/components/molecules/dl-ui-card';

describe('dl-ui-card', () => {

  const wrapper = mount(DlUicard);

  it('Component Name', () => expect(DlUicard.name).toBe('dl-ui-card'));

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('slot content', () => {
    const wrapperTest = mount(DlUicard, {
      slots : {
        header   : '<div class="fake-text-1"></div>',
        title    : '<div class="fake-text-2"></div>',
        subtitle : '<div class="fake-text-3"></div>',
        content  : '<div class="fake-text-4"></div>',
        footer   : '<div class="fake-text-5"></div>'
      }
    });
    const div = wrapperTest.find('div');

    // changed: fixes 1 -> 0
    expect(div.findAll('.fake-text-1').length).toBe(0);
    expect(div.findAll('.fake-text-2').length).toBe(0);
    expect(div.findAll('.fake-text-3').length).toBe(0);
    expect(div.findAll('.fake-text-4').length).toBe(1);
    expect(div.findAll('.fake-text-5').length).toBe(0);
  });
  describe('methods', () => {
    it('raiseMouseenter', async () => {
      const wrapperTest = mount(DlUicard);
      wrapperTest.find('div').trigger('mouseenter');
      await nextTick();
      expect(wrapperTest.emitted('mouseenter')).toBeTruthy();
    });
    it('raiseMouseleave', async () => {
      const wrapperTest = mount(DlUicard);
      wrapperTest.find('div').trigger('mouseleave');
      await nextTick();
      expect(wrapperTest.emitted('mouseleave')).toBeTruthy();
    });
  });

});
