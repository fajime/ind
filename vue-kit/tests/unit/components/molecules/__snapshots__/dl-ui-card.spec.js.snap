// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dl-ui-card renders correctly 1`] = `
<div
  class="dl-ui-card"
>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  <div
    class="dl-ui-card__header"
  >
    <transition-stub>
      <!--v-if-->
    </transition-stub>
    <div
      class="dl-ui-card__title-alt dl-ui-card__title"
    >
      Title
    </div>
    <div
      class="dl-ui-card__icons"
    >
      
      
    </div>
  </div>
  <div
    class="dl-ui-card__body"
  >
    <div
      class="dl-ui-card__content"
    >
      
      
    </div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>
`;
